master/static/migrations/20240508154139_add-custom-project-key.tx.up.sql

```diff
@@ -0,0 +1,33 @@
+CREATE OR REPLACE FUNCTION function_generate_project_key(
+    max_key_length INTEGER,
+    max_prefix_length INTEGER,
+    input_string TEXT
+)
+RETURNS TEXT AS $$
+DECLARE
+    prefix TEXT;
+    prefix_length INT;
+    suffix TEXT;
+BEGIN
+    -- Take the first 3 characters of the input string
+    prefix := UPPER(LEFT(input_string, max_prefix_length));
+    prefix_length := LENGTH(prefix);
+    -- Generate a random suffix
+    suffix := UPPER(LEFT(md5(random()::text), max_key_length - prefix_length));
+    
+    -- Check if the key already exists and loop until we find a unique key
+    WHILE EXISTS(SELECT 1 FROM projects WHERE key = prefix || suffix) LOOP
+       suffix := UPPER(LEFT(md5(random()::text), max_key_length - prefix_length));
+    END LOOP;
+    
+    RETURN prefix || suffix;
+END;
+$$ LANGUAGE plpgsql;
+
+ALTER TABLE projects ADD COLUMN IF NOT EXISTS key VARCHAR(5) UNIQUE;
+
+UPDATE 
+    projects 
+SET 
+    key = function_generate_project_key(5, 3, name)
+WHERE key IS NULL;
```

Comment:
Should we have a migration down file?

Reply:
Nope, down migrations got removed recently 🥳